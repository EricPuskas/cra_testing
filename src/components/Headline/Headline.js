import React, { Component } from "react";
import PropTypes from "prop-types";

class Headline extends Component {
  render() {
    const { header, description } = this.props;
    if (!header) return null;
    return (
      <div data-test="Headline">
        <h1 data-test="header"> {header} </h1>
        <p data-test="description"> {description} </p>
      </div>
    );
  }
}

Headline.propTypes = {
  header: PropTypes.string,
  description: PropTypes.string,
  tempArr: PropTypes.arrayOf(
    PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      age: PropTypes.number,
      email: PropTypes.string,
      onlineStatus: PropTypes.bool
    })
  )
};

export default Headline;
