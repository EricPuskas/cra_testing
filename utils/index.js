import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../src/reducers";
import { middleware } from "../src/store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) =>
  checkPropTypes(component.propTypes, expectedProps, "props", component.name);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
