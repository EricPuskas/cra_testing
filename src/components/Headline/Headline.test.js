import React from "react";
import { shallow } from "enzyme";
import { findByAttribute, checkProps } from "../../../utils";
import Headline from "./Headline";

const setup = (props = {}) => {
  const component = shallow(<Headline {...props} />);
  return component;
};
describe("Headline Component", () => {
  describe("Checking PropTypes", () => {
    it("Should not throw a warning", () => {
      const expectedProps = {
        header: "Test Header",
        description: "Test Description",
        tempArr: [
          {
            firstName: "John",
            lastName: "Doe",
            age: 60,
            email: "johndoe@gmail.com",
            onlineStatus: false
          }
        ]
      };

      const propsErr = checkProps(Headline, expectedProps);
      expect(propsErr).toBeUndefined();
    });
  });

  describe("Have props", () => {
    let wrapper;
    const props = {
      header: "Hello World",
      description: "Lorem Ipsum Dolor"
    };
    beforeEach(() => {
      wrapper = setup(props);
    });
    it("Should render without errors", () => {
      const component = findByAttribute(wrapper, "Headline");
      expect(component.length).toBe(1);
    });

    it("Should render an H1", () => {
      const h1 = findByAttribute(wrapper, "header");
      expect(h1.length).toBe(1);
    });
    it("Should render a paragraph(description)", () => {
      const paragraph = findByAttribute(wrapper, "description");
      expect(paragraph.length).toBe(1);
    });
  });

  describe("Have no props", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = setup();
    });
    it("Should not render H1", () => {
      const h1 = findByAttribute(wrapper, "header");
      expect(h1.length).toBe(0);
    });
  });

  it("Should pass snapshot test", () => {
    const component = setup();
    expect(component.debug()).toMatchSnapshot();
  });
});
