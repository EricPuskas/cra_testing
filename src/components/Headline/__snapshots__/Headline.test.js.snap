// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline Component Should pass snapshot test 1`] = `null`;
